content/blog/2020/how-selenium-works-3-isdiplayed.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "How Selenium Works: Episode 3 - isDisplayed"
+publishDate: 2020-02-27
+---
+
+After an interaction on the last weekend of January 2020, on a Selenium Issue where someone said “why can’t you just…” after I explained the issue I thought that I would start explaining commands in Selenium WebDriver and why we landed on the design that we have today.
+
+I will repeat this on every page of the series but a lot, an annoying amount sometimes, of thinking goes into how every little bit of Selenium works. 
+
+In this episode we are going to look at the vast amount of work that goes into `isDisplayed()`. This work is used in interaction commands so it is good to get an understanding of how this all works and how other commands use it. From a technical point of view, this is my favourite command and how it works.
+
+## Why do we care if it is "displayed"?
+
+So, let's start with understanding why we would even want to care if the element is displayed or not. 
+
+Firstly, testing is one the biggest use cases for Selenium so when people do an interaction on the page. Unfortunately, web pages are designed for showing information and not necessarily designed for interactive web pages. As we move to a more interactive web we needed a tool that could at least try workout if elements are visible. 
+
+There are two main cases that we will try see if an element is visible. When we call the `element.is_displayed()` for looking at an element and when we do interactions.
+
+_NOTE_
+
+_Selenium as 2 types of commands when it comes to interactions. This is not click or type. Selenium has the idea of "do what I mean" and "do what I say". When we do `element.click()` we are doing the former. It is a "do what I mean command" and when we look at Actions, which we will discuss in a future episode, is a "do what I want" as we have a lot more granularity in those APIs._
+
+## How does it work?
+
+First of all, we need to understand that `element.is_displayed()` needs to work without having to make the page scroll. This is important as we don't want to be moving the page unnecessarily.
+
+From there we need to have a look at the element. We need to start looking at the CSS that is on the element. 
+
+First, let's see if the element would still be part of the accessibility tree. We don't look in the accessibility tree, we just look at some scenarios where they won't be there and they won't influence any positioning of elements on the page. We don't look in the accessibility tree as this can be an extremely expensive operation in the browser. 
```

Comment:
Perhaps a link to something that describes the "accessibility tree"? I doubt it something that every reader is familiar with.